fastapi/dependencies/utils.py

```diff
@@ -587,13 +587,25 @@ async def solve_dependencies(
         response = Response()
         del response.headers["content-length"]
         response.status_code = None  # type: ignore
-    dependency_cache = dependency_cache or {}
+    if dependency_cache is None:
+        dependency_cache = {}
     sub_dependant: Dependant
     for sub_dependant in dependant.dependencies:
         sub_dependant.call = cast(Callable[..., Any], sub_dependant.call)
         sub_dependant.cache_key = cast(
             Tuple[Callable[..., Any], Tuple[str]], sub_dependant.cache_key
         )
+
+        if sub_dependant.use_cache:
+            # Use a unique object to compare against in case the cached value is None
+            cache_miss = object()
+            cached_value = dependency_cache.get(sub_dependant.cache_key, cache_miss)
+            # If the sub dependant is already cached, skip doing any more work
+            if cached_value is not cache_miss:
+                if sub_dependant.name is not None:
+                    values[sub_dependant.name] = cached_value
+                continue
```

Comment:
Are there any reasons not to simplify it this way?

```suggestion
            if sub_dependant.cache_key in dependency_cache:
                cached_value = dependency_cache[sub_dependant.cache_key]
                if sub_dependant.name is not None:
                    values[sub_dependant.name] = cached_value
                continue
```

Reply:
`dict.get(...)` is atomic operation, and thus thread safe, while consequent `key in dict; dict[key]` is not

Reply:
Good point!
But currently dependency solving process is not parallel.
Also, it's highly unlikely (useless?) that somebody decide to delete existing key from `dependency_cache`.

I would vote for clarity in this case if nobody brings other arguments

Reply:
We have been running this fork in our production application for quite a while now, and before we changed to an atomic `.get()` we got hard to reproduce KeyErrors.